pkg/asset/installconfig/installconfig.go

```diff
@@ -123,8 +124,12 @@ func (a *InstallConfig) Load(f asset.FileFetcher) (found bool, err error) {
 	}
 
 	config := &types.InstallConfig{}
-	if err := yaml.Unmarshal(file.Data, config); err != nil {
-		return false, errors.Wrapf(err, "failed to unmarshal %s", installConfigFilename)
+	if err := yaml.UnmarshalStrict(file.Data, config, yaml.DisallowUnknownFields); err != nil {
+		err = errors.Wrapf(err, "failed to unmarshal %s", installConfigFilename)
+		if strings.Contains(err.Error(), "unknown field") {
+			err = errors.Wrapf(err, "failed to parse first occurence of unknown field")
```

Comment:
The order of this wrapping is backwards. The "failed to unmarshal" error should wrap around the "failed to parse" error.